pennylane_lightning/core/src/lightning_gpu.toml

```diff
@@ -0,0 +1,118 @@
+schema = 1
+
+[device]
+name = "lightning.qubit"
+precision = ['float32', 'float64']
+
+[operations]
+observables = [
+	"NamedObs",
+	"HermitianObs",
+	"TensorObs",
+	"HamiltonianObs",
+]
+
+# These are gates which are defined in Quantum Instruction Set.
+[[operations.gates]]
+native = [
+        "QubitUnitary",
+        "PauliX",
+        "PauliY",
+        "PauliZ",
+        "MultiRZ",
+        "Hadamard",
+        "S",
+        "T",
+        "CNOT",
+        "SWAP",
+        "CSWAP",
+        "Toffoli",
+        "CY",
+        "CZ",
+        "PhaseShift",
+        "ControlledPhaseShift",
+        "RX",
+        "RY",
+        "RZ",
+        "Rot",
+        "CRX",
+        "CRY",
+        "CRZ",
+        "CRot",
+        "IsingXX",
+        "IsingYY",
+        "IsingZZ",
+        "IsingXY",
+        # Adjoint'ed gates
+        # which just need the inverse flag set
+        # are considered natively supported gates.
+        "Adjoint(S)",
+        "Adjoint(T)",
+]
+# decomp gates are those which are neither supported natively
+# and are not executed via the applyMatrix operation. Technically
+# they could also be decomposed via applyMatrix.
+decomp = [
+        "SX",
+        "ISWAP",
+        "PSWAP",
+        "SISWAP",
+        "SQISW",
+        "CPhase",
+        # Adjoint'ed gates
+        # which are not in native set must either be decomposed
+        # or applied via matrix.
+        "Adjoint(SX)",
+        "Adjoint(ISWAP)",
+        "Adjoint(SISWAP)",
+        "Identity",
+        "BasisState",
+        "QubitStateVector",
+        "StatePrep",
+        "ControlledQubitUnitary",
+        "DiagonalQubitUnitary",
+        "SingleExcitation",
+        "SingleExcitationPlus",
+        "SingleExcitationMinus",
+        "DoubleExcitation",
+        "DoubleExcitationPlus",
+        "DoubleExcitationMinus",
+        "QubitCarry",
+        "QubitSum",
+        "OrbitalRotation",
+        "QFT",
+        "ECR",
+]
+# QubitUnitary should never be here.
+# Gates in "matrix" are decomposed to QubitUnitary.
+# Having a QubitUnitary here would mean an infinite loop.
```

Comment:
Is this something we should fundamentally take care for?

Reply:
I think this is problem stems from the decomposition strategy. By decomposing a QubitUnitary into a QubitUnitary, one creates an infinite loop. This is also something that equality saturation would fix as no new terms would be added to the e-class. A QubitUnitary with params P is equivalent to QubitUnitary with params P